modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -4253,15 +4258,40 @@ def iloc_mut(partition, row_internal_indices, col_internal_indices, item):
                 partition.iloc[row_internal_indices, col_internal_indices] = item.copy()
             return partition
 
+        if not is_scalar(item):
+            broadcasted_item, broadcasted_dtypes = broadcast_item(
+                self,
+                row_numeric_index,
+                col_numeric_index,
+                item,
+                need_columns_reindex=need_columns_reindex,
+            )
+        else:
+            broadcasted_item, broadcasted_dtypes = item, pandas.Series(
+                [np.array(item).dtype] * len(col_numeric_index)
+            )
+
+        new_dtypes = None
+        if (
+            # compute dtypes only if assigning entire columns
+            isinstance(row_numeric_index, slice)
+            and row_numeric_index == slice(None)
```

Comment:
isn't it enough to check `row_numeric_index == slice(None)` without checking for it being a slice?..

Reply:
No.
For example:
```python
(Pdb) row_numeric_index
array([1, 2], dtype=int64)
(Pdb) if row_numeric_index == slice(None): print("hello") 
*** ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```